retrofit/src/main/java/retrofit/ExecutorCallAdapterFactory.java

```diff
@@ -53,8 +53,29 @@
       this.delegate = delegate;
     }
 
-    @Override public void enqueue(Callback<T> callback) {
-      delegate.enqueue(new ExecutorCallback<>(callbackExecutor, callback));
+    @Override public void enqueue(final Callback<T> callback) {
+      delegate.enqueue(new Callback<T>() {
+        @Override public void onResponse(final Response<T> response) {
+          callbackExecutor.execute(new Runnable() {
+            @Override public void run() {
+              if (delegate.isCanceled()) {
```

Comment:
interestingly, this only happens if the cancel is received `delegate.enqueue()` and `run()`. I suspect it'll also work without this case.